plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require("fs");
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+  constructor(extensionContext) {
+    this._extensionContext = extensionContext;
+    this._dataAttributes = {
+      columns: ["timestamp", "file-ref", "event-id", "file-name-or-duration"],
+    };
+    this._tmpLogLines = [];
+    this._fileIndex = 0;
+    this._writeToFileTimer = null;
+    this._timerActivated = false;
+    if (!fs.existsSync(this._extensionContext.logUri.fsPath)) {
+      fs.mkdirSync(this._extensionContext.logUri.fsPath);
+      this.writeHeaderLine();
+    }
+    this._performanceTxtPath = path.join(
+      this._extensionContext.logUri.fsPath,
+      "qljsPerformanceTrace.txt"
+    );
+    this._writableStream = fs.createWriteStream(this._performanceTxtPath, {
+      encoding: "utf8",
+      flags: "a",
+    });
+    this._currentSessionMap = {};
+  }
+
+  fileNameOptimizer(fileName) {
+    if (
+      Object.prototype.hasOwnProperty.call(this._currentSessionMap, fileName)
+    ) {
+      this._fileIndex = this._currentSessionMap[fileName];
+    } else {
+      this._fileIndex++;
```

Comment:
Blocking: This will cause file indexes to be reused. Example:
```js
pw.fileNameOptimizer("a.js");  // this._fileIndex = 1
pw.fileNameOptimizer("b.js");  // this._fileIndex = 2
pw.fileNameOptimizer("a.js");  // this._fileIndex = 1
pw.fileNameOptimizer("c.js");  // this._fileIndex = 2 // but b.js already claimed 2
```

Reply:
Fixed in latest commit